### PR TITLE
revoke user, revoke role, and delete role in auth v3

### PR DESCRIPTION
**Pull request by mitake**
This PR implements RPCs and commands for revoking a role from a user, revoking a key from a role, and deleting role in auth v3.

### CODE REVIEW

File: auth/store.go [hunk 1]
```diff
@@ -76,6 +76,9 @@ type AuthStore interface {
 	// UserGet gets the detailed information of a user
 	UserGet(r *pb.AuthUserGetRequest) (*pb.AuthUserGetResponse, error)
 
+	// UserRevoke revokes a role of a user
+	UserRevoke(r *pb.AuthUserRevokeRequest) (*pb.AuthUserRevokeResponse, error)
```
Comment (xiang90) [inline auth/store.go]: I feel we should probable rename this to userRevokeRole (also userGrantRole) in another pr. userRevoke is easily confused with userDelete. also for roleRevoke -> roleRevokePermission. But it is fine for this pr.

File: auth/store.go [hunk 2]
```diff
@@ -324,6 +333,43 @@ func (as *authStore) UserGet(r *pb.AuthUserGetRequest) (*pb.AuthUserGetResponse,
...
+	if err != nil {
+		return nil, err
+	}
+
+	updatedUser := &authpb.User{}
+	updatedUser.Name = user.Name
+	updatedUser.Password = user.Password
+
+	for _, role := range user.Roles {
+		if strings.Compare(role, r.Role) != 0 {
```
Comment (xiang90) [inline auth/store.go]: maybe we want to return an error if user tries to revoke a non-existing role? (add a todo is ok)

Comment (mitake) [inline auth/store.go]: Ah yes, I'll add the todo.

File: auth/store.go (hunk 1 above)
Comment (mitake) [inline auth/store.go]: UserRevokeRole seems to be better. I'll send another cleanup PR later.

**Comment by mitake** [comment]
@xiang90 updated for the TODO, PTAL. I'll send another PR for the cleanings later.

File: etcdctl/ctlv3/command/role_command.go [hunk 3]
```diff
@@ -60,6 +62,22 @@ func newRoleGetCommand() *cobra.Command {
 	}
 }
 
+func newRoleRevokeCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:   "revoke <role name> <key>",
```
Comment (xiang90) [inline etcdctl/ctlv3/command/role_command.go]: revoke-permission?

File: etcdctl/ctlv3/command/role_command.go [hunk 4]
```diff
@@ -118,3 +136,31 @@ func roleGetCommandFunc(cmd *cobra.Command, args []string) {
...
+	if len(args) != 2 {
+		ExitWithError(ExitBadArgs, fmt.Errorf("role revoke command requires role name and key as its argument."))
+	}
+
+	_, err := mustClientFromCmd(cmd).Auth.RoleRevoke(context.TODO(), args[0], args[1])
+	if err != nil {
+		ExitWithError(ExitError, err)
+	}
+
+	fmt.Printf("Role %s updated\n", args[0])
```
Comment (xiang90) [inline etcdctl/ctlv3/command/role_command.go]: permission x is revoked from role x?

File: etcdctl/ctlv3/command/user_command.go [hunk 5]
```diff
@@ -92,6 +93,14 @@ func newUserGetCommand() *cobra.Command {
 	}
 }
 
+func newUserRevokeCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:   "revoke <user name> <role name>",
```
Comment (xiang90) [inline etcdctl/ctlv3/command/user_command.go]: revoke-role?

**Comment by xiang90** [comment]
LGTM after fixing the few comments.

**Comment by mitake** [comment]
@xiang90 updated for the comments, PTAL.

**Pull request merged**